x/feegrant/client/cli/cli_test.go

```diff
@@ -0,0 +1,481 @@
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+
+		s.Run(tc.name, func() {
+			cmd := cli.NewCmdFeeGrant()
+			out, err := clitestutil.ExecTestCLICmd(clientCtx, cmd, tc.args)
+
+			if tc.expectErr {
+				s.Require().Error(err)
+			} else {
+				s.Require().NoError(err)
+				s.Require().NoError(clientCtx.JSONMarshaler.UnmarshalJSON(out.Bytes(), tc.respType), out.String())
+
+				txResp := tc.respType.(*sdk.TxResponse)
+				s.Require().Equal(tc.expectedCode, txResp.Code, out.String())
+			}
+		})
+	}
+}
+
+func (s *IntegrationTestSuite) TestNewCmdRevokeFeegrant() {
+	val := s.network.Validators[0]
+	granter := s.addedGranter
+	grantee := s.addedGrantee
+	clientCtx := val.ClientCtx
+
+	commonFlags := []string{
+		fmt.Sprintf("--%s=%s", flags.FlagBroadcastMode, flags.BroadcastBlock),
+		fmt.Sprintf("--%s=true", flags.FlagSkipConfirmation),
+		fmt.Sprintf("--%s=%s", flags.FlagFees, sdk.NewCoins(sdk.NewCoin(s.cfg.BondDenom, sdk.NewInt(10))).String()),
+	}
+
+	testCases := []struct {
+		name         string
+		args         []string
+		expectErr    bool
+		respType     proto.Message
+		expectedCode uint32
+	}{
+		{
+			"invalid grantee",
+			append(
+				[]string{
+					"wrong_granter",
+					grantee.String(),
+					fmt.Sprintf("--%s=%s", flags.FlagFrom, granter),
+				},
+				commonFlags...,
+			),
+			true,
+			nil,
+			0,
+		},
+		{
+			"invalid grantee",
+			append(
+				[]string{
+					granter.String(),
+					"wrong_grantee",
+					fmt.Sprintf("--%s=%s", flags.FlagFrom, granter),
+				},
+				commonFlags...,
+			),
+			true,
+			nil,
+			0,
+		},
+		{
+			"Non existed grant",
+			append(
+				[]string{
+					granter.String(),
+					"cosmos1aeuqja06474dfrj7uqsvukm6rael982kk89mqr",
+					fmt.Sprintf("--%s=%s", flags.FlagFrom, granter),
+				},
+				commonFlags...,
+			),
+			false,
+			&sdk.TxResponse{},
+			4,
+		},
+		{
+			"Valid revoke",
+			append(
+				[]string{
+					granter.String(),
+					grantee.String(),
+					fmt.Sprintf("--%s=%s", flags.FlagFrom, granter),
+				},
+				commonFlags...,
+			),
+			false,
+			&sdk.TxResponse{},
+			0,
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+
+		s.Run(tc.name, func() {
+			cmd := cli.NewCmdRevokeFeegrant()
+			out, err := clitestutil.ExecTestCLICmd(clientCtx, cmd, tc.args)
+
+			if tc.expectErr {
+				s.Require().Error(err)
+			} else {
+				s.Require().NoError(err)
+				s.Require().NoError(clientCtx.JSONMarshaler.UnmarshalJSON(out.Bytes(), tc.respType), out.String())
+
+				txResp := tc.respType.(*sdk.TxResponse)
+				s.Require().Equal(tc.expectedCode, txResp.Code, out.String())
+			}
+		})
+	}
+}
+
+func (s *IntegrationTestSuite) TestTxWithFeeGrant() {
+	val := s.network.Validators[0]
+	granter := s.addedGranter
+	grantee := s.addedGrantee
+
+	out, err := govtestutil.MsgSubmitProposal(val.ClientCtx, grantee.String(),
+		"Text Proposal", "No title", govtypes.ProposalTypeText,
+		fmt.Sprintf("--%s=%s", flags.FlagFeeAccount, granter.String()),
+	)
+
+	fmt.Println("out, err", out, err)
+}
```

Comment:
can we add some error check in here?

Reply:
Yeah, facing some issues here with flags debugging this?